flexmeasures/api/v2_0/tests/utils.py

```diff
@@ -17,3 +24,117 @@ def get_asset_post_data() -> dict:
+                     e.g. (assuming tile_n=1) duration=timedelta(hours=6) leads to a resolution of 15 minutes,
+                     and duration=timedelta(hours=48) leads to a resolution of 2 hours.
+    """
+    message = {
+        "type": "PostPriceDataRequest",
+        "market": "ea1.2018-06.localhost:epex_da",
+        "values": tile(
+            [
+                52.37,
+                51.14,
+                49.09,
+                48.35,
+                48.47,
+                49.98,
+                58.7,
+                67.76,
+                69.21,
+                70.26,
+                70.46,
+                70,
+                70.7,
+                70.41,
+                70,
+                64.53,
+                65.92,
+                69.72,
+                70.51,
+                75.49,
+                70.35,
+                70.01,
+                66.98,
+                58.61,
+            ],
+            tile_n,
+        ).tolist(),
+        "start": "2021-01-06T00:00:00+01:00",
+        "duration": duration_isoformat(timedelta(hours=24 * tile_n)),
+        "horizon": duration_isoformat(timedelta(hours=0)),
+        "unit": "EUR/MWh",
+    }
+    if duration is not None:
+        message["duration"] = duration
+    if compress_n > 1:
+        message["values"] = message["values"][::compress_n]
+    if invalid_unit:
+        message["unit"] = "KRW/kWh"  # That is, an invalid unit for EPEX SPOT.
+    if no_horizon or prior_instead_of_horizon:
+        message.pop("horizon", None)
+    if prior_instead_of_horizon:
+        message["prior"] = "2021-01-05T12:00:00+01:00"
+    return message
+
+
+def verify_prices_in_db(post_message, values, db, swapped_sign: bool = False):
+    """util method to verify that price data ended up in the database"""
+    start = parse_datetime(post_message["start"])
+    end = start + parse_duration(post_message["duration"])
+    market: Market = get_market(post_message)
+    resolution = market.event_resolution
+    if "horizon" in post_message:
+        horizon = parse_duration(post_message["horizon"])
+        query = (
+            db.session.query(Price.datetime, Price.value, Price.horizon)
+            .filter((Price.datetime > start - resolution) & (Price.datetime < end))
+            .filter(Price.horizon == horizon)
+            .join(Market)
+            .filter(Market.name == market.name)
+        )
+    else:
+        query = (
+            db.session.query(
+                Price.datetime,
+                Price.value,
+                Price.horizon,
+            )
+            .filter((Price.datetime > start - resolution) & (Price.datetime < end))
+            # .filter(Price.horizon == (Price.datetime + resolution) - prior)  # only for sensors with 0-hour ex_post knowledge horizon function
+            .join(Market)
+            .filter(Market.name == market.name)
+        )
+    df = pd.DataFrame(
```

Comment:
Maybe a TODO here, so we can get a BeliefsDataFrame from querying directly later?